infrasim/vm.py

```diff
@@ -101,115 +107,109 @@ def set_network(self, mode, name):
         nets = []
         change_flag = False
 
-        conf = cfgparse.ConfigParser()
-        f = conf.add_file(VM_DEFAULT_CONFIG, type = 'ini')
-        conf.parse()
+        with open(VM_DEFAULT_CONFIG, 'r') as f_yml:
+            conf = yaml.load(f_yml)
+
+        # conf = cfgparse.ConfigParser()
+        # f = conf.add_file(VM_DEFAULT_CONFIG, type = 'ini')
+        # conf.parse()
 
         # add existing options into config parser
-        if self.has_option(conf, "node", "qemu_mac_num") is True:
-            conf.add_option("qemu_mac_num", keys="node")
+
+        # if has_option(conf, "node", "qemu_mac_num") is True:
+        #     conf.add_option("qemu_mac_num", keys="node")
 
         mac_list = []
         for i in range(0, MAX_QEMU_MAC_NUM):
-            if self.has_option(conf, "node", "qemu_mac"+str(i)) is True:
-                mac_list.append(conf.add_option("qemu_mac"+str(i), keys="node"))
+            if has_option(conf, "node", "qemu_mac"+str(i)):
+                mac_list.append(conf["node"]["qemu_mac"+str(i)])
 
-        # parse valid options
-        opts = conf.parse()
+        # # parse valid options
+        # opts = conf.parse()
 
-        #set mac number
-        if self.has_option(conf, "node", "qemu_mac_num") is True:
-            mac_num = opts.qemu_mac_num
+        # set mac number
+        if has_option(conf, "node", "qemu_mac_num"):
+            mac_num = conf["node"]["qemu_mac_num"]
         else:
             mac_num = 1
 
-        #set netmode in vnode.xml
+        # set netmode in vnode.xml
         if mode == "nat":
             self.node["netmode"] = "nat"
         else:
             self.node["netmode"] = "bridge"
 
-        #set mac address
+        # set mac address
         for i in range(0,int(mac_num)):
-            #init mac, read mac from config or generate random mac
+            # init mac, read mac from config or generate random mac
             mac = "null"
-            if self.has_option(conf, "node", "qemu_mac"+str(i)) is True:
-                mac = mac_list[i].get()
+            if has_option(conf, "node", "qemu_mac"+str(i)):
+                mac = mac_list[i]
                 if mac == "null":
                     mac = self.mac_generator()
-                    f.set_option("qemu_mac"+str(i), mac, "node")
+                    conf["node"]["qemu_mac"+str(i)] = mac
```

Comment:
named with "network_mac1", not qemu_mac1 in yml file. naming with qemu_mac1 sounds better